src/pgsql.cpp

```diff
@@ -55,11 +55,18 @@ void pg_conn_t::set_config(char const *setting, char const *value) const
     query(PGRES_TUPLES_OK, sql);
 }
 
-void pg_conn_t::exec(char const *sql) const { query(PGRES_COMMAND_OK, sql); }
+void pg_conn_t::exec(char const *sql) const
+{
+    if (sql && sql[0] != '\0') {
+        query(PGRES_COMMAND_OK, sql);
+    }
+}
 
 void pg_conn_t::exec(std::string const &sql) const
 {
-    query(PGRES_COMMAND_OK, sql.c_str());
+    if (!sql.empty()) {
+        query(PGRES_COMMAND_OK, sql.c_str());
+    }
```

Comment:
These two changes don't see really needed for this PR?

Reply:
Because of [this `if`](https://github.com/openstreetmap/osm2pgsql/blob/cd9770b2d724f4e7602b50659acc6684d86d3a51/src/middle-pgsql.cpp#L634), `sql.create_table` and `sql.prepare_query` can be empty, but they are still run. It was easiest to just not run empty queries instead of putting checks in more places.

For consistency I decided to add the emptyness check in both `exec` functions.